src/main/java/com/intel/gkl/compression/IntelInflater.java

```diff
@@ -119,6 +127,9 @@ public void setInput(byte[] b, int off, int len) throws NullPointerException {
         if(b == null) {
             throw new NullPointerException("Input is null");
         }
+        if(off < 0 || off > b.length - len) {
```

Comment:
Can we check for null before value check? Also for value condition can use IllegalArgumentException to be consistent.

Reply:
primitive data types cannot be null since they have a default value